sql/core/src/test/scala/org/apache/spark/sql/jdbc/v2/JDBCV2JoinPushdownIntegrationSuiteBase.scala

```diff
@@ -0,0 +1,522 @@
+      val surname = s"surname_$i"
+      (id, next_id, salary, surname)
+    }
+
+    val insertStmt1 = connection.prepareStatement(
+      s"INSERT INTO $fullyQualifiedTableName1 (id, amount, address) VALUES (?, ?, ?)"
+    )
+    table1Data.foreach { case (id, amount, address) =>
+      insertStmt1.setInt(1, id)
+      insertStmt1.setBigDecimal(2, amount.bigDecimal)
+      insertStmt1.setString(3, address)
+      insertStmt1.executeUpdate()
+    }
+    insertStmt1.close()
+
+    val insertStmt2 = connection.prepareStatement(
+      s"INSERT INTO $fullyQualifiedTableName2 (id, next_id, salary, surname) VALUES (?, ?, ?, ?)"
+    )
+    table2Data.foreach { case (id, next_id, salary, surname) =>
+      insertStmt2.setInt(1, id)
+      insertStmt2.setInt(2, next_id)
+      insertStmt2.setBigDecimal(3, salary.bigDecimal)
+      insertStmt2.setString(4, surname)
+      insertStmt2.executeUpdate()
+    }
+    insertStmt2.close()
+  }
+
+  // Condition-less joins are not supported in join pushdown
+  test("Test that 2-way join without condition should not have join pushed down") {
+    val sqlQuery =
+      s"""
+         |SELECT * FROM
+         |$catalogAndNamespace.${caseConvert(joinTableName1)} a,
+         |$catalogAndNamespace.${caseConvert(joinTableName1)} b
+         |""".stripMargin
+
+    val rows = withSQLConf(SQLConf.DATA_SOURCE_V2_JOIN_PUSHDOWN.key -> "false") {
+      sql(sqlQuery).collect().toSeq
+    }
+
+    withSQLConf(SQLConf.DATA_SOURCE_V2_JOIN_PUSHDOWN.key -> "true") {
+      val df = sql(sqlQuery)
+
+      checkJoinNotPushed(df)
+      checkAnswer(df, rows)
+    }
+  }
+
+  // Condition-less joins are not supported in join pushdown
+  test("Test that multi-way join without condition should not have join pushed down") {
+    val sqlQuery = s"""
+      |SELECT * FROM
+      |$catalogAndNamespace.${caseConvert(joinTableName1)} a,
+      |$catalogAndNamespace.${caseConvert(joinTableName1)} b,
+      |$catalogAndNamespace.${caseConvert(joinTableName1)} c
+      |""".stripMargin
+
+    val rows = withSQLConf(SQLConf.DATA_SOURCE_V2_JOIN_PUSHDOWN.key -> "false") {
+      sql(sqlQuery).collect().toSeq
+    }
+
+    withSQLConf(SQLConf.DATA_SOURCE_V2_JOIN_PUSHDOWN.key -> "true") {
+      val df = sql(sqlQuery)
+
+      checkJoinNotPushed(df)
+      checkAnswer(df, rows)
+    }
+  }
+
+  test("Test self join with condition") {
+    val sqlQuery = s"""
+      |SELECT * FROM $catalogAndNamespace.${caseConvert(joinTableName1)} a
+      |JOIN $catalogAndNamespace.${caseConvert(joinTableName1)} b
+      |ON a.id = b.id + 1""".stripMargin
+
+    val rows = withSQLConf(SQLConf.DATA_SOURCE_V2_JOIN_PUSHDOWN.key -> "false") {
+      sql(sqlQuery).collect().toSeq
+    }
+
+    withSQLConf(SQLConf.DATA_SOURCE_V2_JOIN_PUSHDOWN.key -> "true") {
+      val df = sql(sqlQuery)
+
+      checkJoinPushed(
+        df,
+        s"$catalogAndNamespace.${caseConvert(joinTableName1)}, " +
+          s"$catalogAndNamespace.${caseConvert(joinTableName1)}"
```

Comment:
You can provide here sequence of relations instead of doing string concat, it would be better fit for your signature.
Also, can you name parameter here?
```suggestion
        expectedTables = Seq(
          s"$catalogAndNamespace.${caseConvert(joinTableName1)}",
          s"$catalogAndNamespace.${caseConvert(joinTableName1)}"
        )
```